src/components/project-input.ts

```diff
@@ -1,74 +1,129 @@
+      let errorText = "";
+      if (!titleIsValid) {
+        if (titleIsRequired && enteredTitle === "") {
+          errorText += `Title is required \n`;
+        }
+        if (titleIsRequired && enteredTitle !== "") {
+          if (titleMinLength > enteredTitle.length) {
+            errorText += `Title needs to have a minimum of ${titleValidatable.minLength} characters\n`;
+          }
+          if (titleMaxLength < enteredTitle.length) {
+            errorText += `Title can have a maximum of ${titleValidatable.maxLength} characters\n`;
+          }
+        }
       }
-    }
-
-    private clearUserInputs() {
-      this.titleInputElement.value = "";
-      this.descriptionInputElement.value = "";
-      this.peopleInputElement.value = "";
-    }
-    private gatherUserInput(): [string, string, number] | void {
-      const enteredTitle = this.titleInputElement.value;
-      const enteredDescription = this.descriptionInputElement.value;
-      const enteredPeoples = this.peopleInputElement.value;
-      const titleValidatable: Validatable = {
-        value: enteredTitle,
-        required: true,
-      };
-      const descriptionValidatable: Validatable = {
-        value: enteredDescription,
-        required: true,
-        minLength: 7,
-      };
-      const peopleValidatable: Validatable = {
-        value: +enteredPeoples,
-        required: true,
-        minValue: 1,
-      };
-      if (
-        !validate(titleValidatable) ||
-        !validate(descriptionValidatable) ||
-        !validate(peopleValidatable)
-      ) {
-        alert("Please enter valid input");
-      } else {
-        return [enteredTitle, enteredDescription, Number(enteredPeoples)];
+      if (!descriptionIsValid) {
+        if (descriptionIsRequired && enteredDescription === "") {
+          errorText += `Description is required \n`;
+        }
+        if (descriptionIsRequired && enteredDescription!=="") {
+          if (descriptionMinLength > enteredDescription.length) {
+            errorText += `Description needs to have a minimum of ${descriptionValidatable.minLength} characters\n`;
+          }
+          if (descriptionMaxLength < enteredDescription.length) {
+            errorText += `Description can have a maximum of ${descriptionValidatable.maxLength} characters\n`;
+          }
+        }
+      }
+      if (!peopleIsValid) {
+        if (peopleIsRequired && enteredPeoples === ""){
+          errorText += `Number of people is required \n`;
+        }
+        if(peopleIsRequired && enteredPeoples!=="") {
+        if (peopleMinValue>+enteredPeoples) {
+          errorText += `There needs to be a minimum of ${peopleValidatable.minValue} people\n`;
+        }
+        if (peopleMaxValue<+enteredPeoples) {
+          errorText += `There can be a maximum of ${peopleValidatable.maxValue} people\n`;
+        }}
       }
+      alert(errorText);
+    } else {
+      return [enteredTitle, enteredDescription, Number(enteredPeoples)];
```

Comment:
if you are trying to cast "enteredPeoples" ( which has a typo btw) this is probably not the best way. Check this article https://stackoverflow.com/questions/14667713/how-to-convert-a-string-to-number-in-typescript

Reply:
as i can see you are doing it with a "+" on the previous lines. IF you are not sure which is the best way to do something check online or consult a colleague. Doing it in different ways is usually not good